contracts/utils/math/SignedMath.sol

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+/**
+ * @dev Standard signed math utilities missing in the Solidity language.
+ */
+library SignedMath {
+    /**
+     * @dev Returns the largest of two signed numbers.
+     */
+    function max(int256 a, int256 b) internal pure returns (int256) {
+        return a >= b ? a : b;
+    }
+
+    /**
+     * @dev Returns the smallest of two signed numbers.
+     */
+    function min(int256 a, int256 b) internal pure returns (int256) {
+        return a < b ? a : b;
+    }
+
+    /**
+     * @dev Returns the average of two signed numbers without overflow.
+     * The result is rounded towards zero.
+     */
+    function average(int256 a, int256 b) internal pure returns (int256) {
+        // Formula from the book "Hacker's Delight"
+        int256 x = (a & b) + ((a ^ b) >> 1);
+        return x + (int256(uint256(x) >> 255) & (a ^ b));
+    }
+
+    /**
+     * @dev Returns the ceiling of the division of two signed numbers.
+     *
+     * This differs from standard division with `/` in that it rounds up instead
+     * of rounding down.
```

Comment:
The language of "rounding up" vs "down" was copied from unsigned math but we need to be careful here. Are we sure we're rounding up? Or are we rounding away from zero, i.e. towards infinity of the same sign?

Reply:
This is "rounding toward infinity of the same sign instead of rounding toward zero".

Maybe ceilDiv is not the right name. Maybe we don't want this function at all, and we want both a real ceilDiv and a floorDiv